Trakt/Api/TraktApi.cs

```diff
@@ -1080,6 +1100,60 @@ private async Task<T> GetFromTrakt<T>(string url, TraktUser traktUser, Cancellat
         }
     }
 
+    private Task<List<T>> GetFromTraktWithPaging<T>(string url, TraktUser traktUser)
+    {
+        return GetFromTraktWithPaging<T>(url, traktUser, CancellationToken.None);
+    }
+
+    private async Task<List<T>> GetFromTraktWithPaging<T>(string url, TraktUser traktUser, CancellationToken cancellationToken)
+    {
+        var httpClient = GetHttpClient();
+        var page = 1;
+        var result = new List<T>();
+
+        if (traktUser != null)
+        {
+            await SetRequestHeaders(httpClient, traktUser).ConfigureAwait(false);
+        }
+
+        await _traktResourcePool.WaitAsync(cancellationToken).ConfigureAwait(false);
+
+        try
+        {
+            while (true)
+            {
+                var urlWithPage = url.Replace("{page}", page.ToString(CultureInfo.InvariantCulture), StringComparison.InvariantCulture);
+                var response = await RetryHttpRequest(async () => await httpClient.GetAsync(urlWithPage, cancellationToken).ConfigureAwait(false)).ConfigureAwait(false);
+                if (response.StatusCode == HttpStatusCode.NotFound)
+                {
+                    return result;
+                }
+
+                response.EnsureSuccessStatusCode();
+                var tmpResult = await response.Content.ReadFromJsonAsync<List<T>>(_jsonOptions, cancellationToken).ConfigureAwait(false);
+                if (result.GetType().IsGenericType && result.GetType().GetGenericTypeDefinition() == typeof(List<>))
```

Comment:
What is this type check for? Could you just check if it's `null` instead?

Reply:
You're right, probably a leftover from some trial before.
Will change to this, ok?
![image](https://user-images.githubusercontent.com/1945295/205506665-fc5b3e14-7d5c-41be-9272-03ac24a0dd84.png)

Reply:
Requested change committed.